ion/binaryreader.go

```diff
@@ -248,192 +259,34 @@ func (r *binaryReader) readBVM() error {
 	}
 }
 
-// ReadLocalSymbolTable reads and installs a new local symbol table.
-func (r *binaryReader) readLocalSymbolTable() error {
-	if r.IsNull() {
-		r.clear()
-		r.lst = V1SystemSymbolTable
-		return nil
-	}
-
-	if err := r.StepIn(); err != nil {
+// ReadFieldName reads and resolves a field name.
+func (r *binaryReader) readFieldName() error {
+	id, err := r.bits.ReadFieldID()
+	if err != nil {
 		return err
 	}
 
-	var imps []SharedSymbolTable
-	var syms []string
-
-	for r.Next() {
-		var err error
-		switch *r.FieldName() {
-		case "imports":
-			imps, err = r.readImports()
-		case "symbols":
-			syms, err = r.readSymbols()
-		}
-		if err != nil {
-			return err
-		}
-	}
-
-	if err := r.StepOut(); err != nil {
+	st, err := r.readSymbol(id)
+	if err != nil {
 		return err
 	}
-
-	r.lst = NewLocalSymbolTable(imps, syms)
+	r.fieldNameSymbol = st
 	return nil
-}
-
-// ReadImports reads the imports field of a local symbol table.
-func (r *binaryReader) readImports() ([]SharedSymbolTable, error) {
-	if r.valueType == SymbolType && r.value == "$ion_symbol_table" {
-		// Special case that imports the current local symbol table.
-		if r.lst == nil || r.lst == V1SystemSymbolTable {
-			return nil, nil
-		}
-
-		imps := r.lst.Imports()
-		lsst := NewSharedSymbolTable("", 0, r.lst.Symbols())
-		return append(imps, lsst), nil
-	}
-
-	if r.Type() != ListType || r.IsNull() {
-		return nil, nil
-	}
-	if err := r.StepIn(); err != nil {
-		return nil, err
-	}
 
-	var imps []SharedSymbolTable
-	for r.Next() {
-		imp, err := r.readImport()
-		if err != nil {
-			return nil, err
-		}
-		if imp != nil {
-			imps = append(imps, imp)
-		}
-	}
-
-	err := r.StepOut()
-	return imps, err
 }
 
-// ReadImport reads an import definition.
-func (r *binaryReader) readImport() (SharedSymbolTable, error) {
-	if r.Type() != StructType || r.IsNull() {
-		return nil, nil
-	}
-	if err := r.StepIn(); err != nil {
-		return nil, err
+// ReadSymbol reads an ID and returns a symbol token.
+func (r *binaryReader) readSymbol(id uint64) (SymbolToken, error) {
+	if id > r.SymbolTable().MaxID() {
+		return symbolTokenUndefined, &UsageError{"Reader.Next", "sid is out of range "}
```

Comment:
I would add the value for `id` and `MaxID()` in the error message

Reply:
Updated